src/controllers/admin/survey-administration-controller.ts

```diff
@@ -0,0 +1,176 @@
+ */
+export const show = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const survey = await SurveyAdministrationService.getById(parseInt(id))
+    res.json(survey)
+  } catch (error) {
+    res.status(500).json({ message: "Something went wrong" })
+  }
+}
+
+/**
+ * Store a new survey administration
+ * @param req.body.name - Name
+ * @param req.body.survey_start_date - Survey start date
+ * @param req.body.survey_end_date - Survey end date
+ * @param req.body.survey_template_id - Survey template id
+ * @param req.body.remarks - Survey remarks
+ * @param req.body.email_subject - Email subject
+ * @param req.body.email_content - Email content
+ * @param req.body.status - Status
+ */
+export const store = async (req: Request, res: Response) => {
+  try {
+    const {
+      name,
+      survey_start_date,
+      survey_end_date,
+      survey_template_id,
+      remarks,
+      email_subject,
+      email_content,
+      status,
+    } = req.body
+    await createSurveyAdministrationSchema.validate({
+      name,
+      survey_start_date,
+      survey_end_date,
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+      status,
+    })
+
+    const newSurvey = await SurveyAdministrationService.create({
+      name: name as string,
+      survey_start_date: new Date(survey_start_date),
+      survey_end_date: new Date(survey_end_date),
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+      status: SurveyAdministrationStatus.Draft,
+    })
+    return res.json(newSurvey)
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return res.status(500).json(error)
+    }
+    if (error instanceof CustomError) {
+      return res.status(error.status).json({ message: error.message })
+    }
+    res.status(500).json({ message: "Something went wrong" })
+  }
+}
+
+/**
+ * Update an existing survey administration by ID
+ * @param req.body.name - Name
+ * @param req.body.survey_start_date - Survey start date
+ * @param req.body.survey_end_date - Survey end date
+ * @param req.body.survey_template_id - Survey template id
+ * @param req.body.remarks - Survey remarks
+ * @param req.body.email_subject - Email subject
+ * @param req.body.email_content - Email content
+ * @param req.body.status - Status
+ */
+export const update = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const {
+      name,
+      survey_start_date,
+      survey_end_date,
+      survey_template_id,
+      remarks,
+      email_subject,
+      email_content,
+      status,
+    } = req.body
+    await createSurveyAdministrationSchema.validate({
+      name,
+      survey_start_date,
+      survey_end_date,
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+      status,
```

Comment:
I think status should not be manually updated here in this api, we can remove status here